l10n_it_split_payment/models/account.py

```diff
@@ -60,70 +64,40 @@ def _build_debit_line(self):
-                inv_total = invoice.amount_sp + invoice.amount_total
-                if invoice.type == "out_invoice":
-                    if inv_total:
-                        receivable_line_amount = (
-                            invoice.amount_total * receivable_line.debit
-                        ) / inv_total
-                    else:
-                        receivable_line_amount = 0
-                    receivable_line.with_context(check_move_validity=False).write(
-                        {"debit": receivable_line_amount}
-                    )
-                elif invoice.type == "out_refund":
-                    if inv_total:
-                        receivable_line_amount = (
-                            invoice.amount_total * receivable_line.credit
-                        ) / inv_total
-                    else:
-                        receivable_line_amount = 0
-                    receivable_line.with_context(check_move_validity=False).write(
-                        {"credit": receivable_line_amount}
-                    )
-
-    @api.multi
-    def action_move_create(self):
-        res = super(AccountInvoice, self).action_move_create()
-        for invoice in self:
-            if invoice.split_payment:
-                if invoice.type in ["in_invoice", "in_refund"]:
-                    raise UserError(
-                        _("Can't handle supplier invoices with split payment")
-                    )
-                if invoice.move_id.state == "posted":
-                    posted = True
-                    invoice.move_id.state = "draft"
-                self._compute_split_payments()
-                line_model = self.env["account.move.line"]
-                write_off_line_vals = invoice._build_debit_line()
-                write_off_line_vals["move_id"] = invoice.move_id.id
-                line_model.with_context(check_move_validity=False).create(
-                    write_off_line_vals
+        write_off_line_vals = self._build_debit_line()
+        line_sp = self.line_ids.filtered(lambda l: l.is_split_payment)
+        if line_sp:
+            dif_price = write_off_line_vals["price_unit"] - line_sp.price_unit
+            if write_off_line_vals["price_unit"] == 0:
+                line_sp.with_context(check_move_validity=False).unlink()
+            else:
+                line_sp.with_context(check_move_validity=False).write(
+                    {"price_unit": write_off_line_vals["price_unit"]}
                 )
-                if posted:
-                    invoice.move_id.state = "posted"
-        return res
+            line_client = self.line_ids.filtered(
+                lambda l: l.account_id.id
+                == self.partner_id.property_account_receivable_id.id
+            )
+            if line_client:
+                line_client.write({"price_unit": line_client.price_unit - dif_price})
+        else:
+            if self.id:
```

Comment:
Come mai hai aggiunto questo controllo? Non ho trovato nulla di simile nel codice pre-migrazione.

Reply:
Se la fattura è in creazione ma non salvata va in errore perché come id ha `idNew`

Reply:
Credo sia rischioso distinguere così il comportamento tra quando la fattura è in creazione rispetto a quando è già creata perché ci si aspetta che il comportamento sia lo stesso.
Potresti invece togliere il campo `move_id` dalla riga di _write off_ e assegnarla direttamente al campo `self.invoice_line_ids`, facendo ad esempio
```suggestion
            self.invoice_line_ids = [(0, 0, write_off_line_vals)]
```
o qualcosa di simile, che ne pensi?

Reply:
grazie per il suggerimento